src/plugin_system/plugin_manager.py

```
@@ -91,6 +92,32 @@ def __init__(self, plugins_dir: str = "plugins",
+        self._pending_updates: set = set()
+        self._pending_lock = threading.Lock()
+        self._plugin_locks: Dict[str, threading.Lock] = {}
+        self._plugin_locks_guard = threading.Lock()
+        self._update_worker: Optional[threading.Thread] = None
+        self._synchronous_updates = False
+        try:
+            if self.config_manager is not None:
+                cfg = self.config_manager.get_config() or {}
+                self._synchronous_updates = bool(
+                    cfg.get('plugin_system', {}).get('synchronous_updates', False))
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)